### PR TITLE
GEOMESA-1480 KafkaDataStore - fixing spatial queries against non-default geometries

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by jnh5y** [comment]
@elahrvivaz this looks good to me.

@mdzimmerman can you read through briefly and comment?  Also, if you have a chance, can you check out the downstream projects which might be affected by the 'getIterator' -> 'toIterator' change?

**Comment by mdzimmerman** [comment]
Doesn't affect Iktomi, which uses the public GeoTools interface. Checking geomesa-tutorials next.

**Comment by mdzimmerman** [comment]
geomesa-tutorials works fine too.

File: geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/index/SpatialIndexSupportTest.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+@RunWith(classOf[JUnitRunner])
+class SpatialIndexSupportTest extends Specification {
+
+  import org.locationtech.geomesa.utils.geotools.Conversions._
+
+  "QuadTreeFeatureStore" should {
```
Comment (elahrvivaz) [inline geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/index/SpatialIndexSupportTest.scala]: fixed

**Comment by jahhulbert-ccri** [comment]
looks good

**Comment by jahhulbert-ccri** [comment]
@elahrvivaz mind rebasing this to remove the conflict?

**Comment by elahrvivaz** [comment]
merge conflict resolved

**Pull request merged**